course-design.md

```diff
@@ -0,0 +1,214 @@
+    - Vignettes, README (with rmarkdown and usethis)
+    - Function documentation (with roxygen2), part 2
+    - learnr tutorials via `use_tutorials()`?
+    - Running spell checks and styling (with spelling and styler)
+
+- Exposing your package to the world with a website:
+    - Setting up a website (with pkgdown)
+    - Customizing the themes
+    - Getting all resources and material organized
+    - Defining lifecycle of your package and individual functions (with lifecycle)
+
+- Involving the community:
+    - Designing your package to be used by or contributed to from the community
+    (refer to Mozilla Science Labs Open Leaders material)
+    - Contributing guidelines 
+    - Being inclusive (code of conduct)
+    - Licensing
+    - Governance
+    - NEWS file
+
+- Version control: Workflows around using GitHub (part 2)
+    - Make use of R builtin tools like the usethis pull request helpers 
+    (and learning how to make use of this yourself)
+    - Using this workflow as a team or as a contributor
+
+- Project management of your package development (either sole or in a team)
+    - Issue tracking
+    - Labeling
+    - Project boards
+    - Milestones
+
+- Preparing for package release
+    - Check builds on other systems (with rhub)
+    - Running local CRAN checks
+    - Managing your versions, git tagging
+
+- R package development in a team-based environment
+    - Making use of GitHub Organizations/Teams
+    - Branch protection, member roles
+    - Resolving conflict, personality differences
+    - Running meetings, code of conduct
+    - Decision making
+    - Governance
+    - Issue assignment
+    - Relying on and using code of conduct to build culture and standards
+
+## Final projects for formal assessment
+
+Two project deliverables for learners:
+
+- Package created alone as the first deliverable
+    - Due after "package release" lesson
+    - This will be a simpler package compared to the next one
+- Package created as a team as the second deliverable
+    - Due after teamwork lesson (so they have learned the basic skills of package dev
+    and now start from scratch with these skills to create a package)
+- Another idea for an additional deliverable could be to submit simple pull
+requests to existing packages to address an issue
+    
+Ideas to how project should be started:
+
+- We give them several general problems to solve, with or without data that needs to be worked
+    - Or existing solutions that could be improved on
+- We let them decide on the problem (this is tricky since it requires domain knowledge)
+- We get them to re-create an existing package that could use more work or development
+(there are no end to these types of packages, especially the earlier ones)
+    - Could get them to refactor code to be more readable
+- We get them to choose an existing package to develop on their own and then
+to contribute back (e.g. develop a website for the package and update the documentation)
```

Comment:
I think we should provide pretty concrete problems/examples, and then let individual instructors deviate to more student directed problems if they want.  E.g. it's a lot easier for an instructor to evaluate an alternate project, if they already have an example to look at that they know will work out.